enemy-test/src/enemy_monster.c

```diff
@@ -0,0 +1,114 @@
+#include <gb/gb.h>
+
+#include "enemy_monster.h"
+#include "dummy.sprites.h"
+
+// Initialize an enemy unit
+void init_enemy(ENEMY* unit, UINT8 enemy_sprite_l, UINT8 enemy_sprite_r, UINT8 s_id, UINT8 attack_type, UINT8 damage, UINT8 hp, UINT8 frames_between_attacks) {
+	unit->enemy_sprite_l = enemy_sprite_l;
+	unit->enemy_sprite_r = enemy_sprite_r;
+	unit->sprite_id = s_id;
+	unit->attack_type = attack_type;
+	unit->max_health = hp;
+	unit->health = hp; // Enemies always start full HP.
+	unit->damage = damage;
+	unit->frames_between_attacks = frames_between_attacks;
+	unit->frames_until_next_attack = frames_between_attacks; // There is one full cycle before the enemy starts behaving normally. Might want to be able to configure that.
+}
+
+// Display enemy unit on-screen at specified x and y coordinates
+void display_enemy(ENEMY* unit, UINT8 xpos, UINT8 ypos) {
+	// Store current position
+	unit->xpos = xpos;
+	unit->ypos = ypos;
+	
+	// Initialize left sprite
+	set_sprite_tile(unit->sprite_id, unit->enemy_sprite_l);
+	move_sprite(unit->sprite_id, xpos, ypos);
+	
+	// Initialize right sprite
+	set_sprite_tile(unit->sprite_id + 1, unit->enemy_sprite_r);
+	move_sprite(unit->sprite_id + 1, xpos + 8, ypos);

```

Comment:
FIXME: Wrong offsets, the gameboy automatically adds (8, 16) to sprites, you need to offset that